spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,371 @@
-      end
-
-      context 'with upcoming and overdue reservations' do
-        before do
-          FactoryGirl.create(:valid_reservation,
-                             equipment_model: @model,
-                             start_date: Time.zone.today + 1.day,
-                             due_date: Time.zone.today + 1.day)
-          FactoryGirl.create(:overdue_reservation,
-                             equipment_model: @model,
-                             start_date: Time.zone.today - 1.day,
-                             due_date: Time.zone.today - 1.day,
-                             equipment_item: @model.equipment_items.first)
-          # needed to persist overdue_count
-          @model.save
+        context 'without @category set' do
+          it 'populates an array of all equipment models' do
+            models = mock_eq_collection
+            expect(models).to receive(:includes)
+            expect(models).not_to receive(:active)
+            get :index, show_deleted: true
+          end
         end
-
-        it_behaves_like 'calculates availability correctly', 0
       end
     end
+    USER_ROLES.each { |type| it_behaves_like 'GET index success', type }
   end
 
   describe 'GET new' do
     context 'with admin user' do
       before do
-        sign_in FactoryGirl.create(:admin)
+        mock_user_sign_in(mock_user(:admin))
         get :new
       end
-      it { is_expected.to respond_with(:success) }
-      it { is_expected.to render_template(:new) }
-      it { is_expected.not_to set_flash }
+      it_behaves_like 'successful request', :new
       it 'assigns a new equipment model to @equipment_model' do
         expect(assigns(:equipment_model)).to be_new_record
         expect(assigns(:equipment_model)).to be_kind_of(EquipmentModel)
       end
-      it 'sets equipment_model to nil when no category is specified' do
-        expect(assigns(:equipment_model).category).to be_nil
-      end
       it 'sets category when one is passed through params' do
-        cat = model.category
-        get :new, category_id: cat
-        expect(assigns(:equipment_model).category).to eq(cat)
+        cat = mock_category(id: 1)
+        expect(EquipmentModel).to receive(:new).with(category: cat)
+        get :new, category_id: cat.id
       end
     end
-    context 'with non-admin user' do
-      before { sign_in FactoryGirl.create(:user) }
-      it 'should redirect to root' do
+    context 'when not admin' do
+      before do
+        mock_user_sign_in
         get :new
-        expect(response).to redirect_to(root_url)
       end
+      it_behaves_like 'redirected request'
     end
   end
 
-  describe 'GET edit' do
+  describe 'POST create' do
     context 'with admin user' do
-      before do
-        sign_in FactoryGirl.create(:admin)
-        get :edit, id: model
+      before { mock_user_sign_in(mock_user(:admin)) }
+      context 'successful save' do
+        let!(:model) { FactoryGirl.build_stubbed(:equipment_model) }
```

Comment:
have to use `build_stubbed` because otherwise the `redirect_to` fails in the controller